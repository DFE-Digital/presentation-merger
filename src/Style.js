/** @module Style */
import XMLFileBase from './XMLFileBase';
// import { extractArray } from './utils';

/**
 * Class prepresenting `style.xml` in the presentation document
 */
export default class Style extends XMLFileBase {
  /**
   * Create a style representation
   * @param {string} xml XML content
   * @param {array} manifest The document manifest of image assets
   * @param {string} id unique identifier for the presentation
   */
  constructor(xml, manifest, id) {
    super(xml, id, 'office:document-styles', manifest);
  }

  /*
  get arrayNotation() {
    return [
      'style:handout-master',
      'style:master-page',
      'style:style',
      'style:page-layout',
      'style:presentation-page-layout',
      'style:default-style',
      'draw:layer',
      'draw:stroke-dash',
      'draw:marker',
      'draw:fill-image',
      'draw:hatch',
      'draw:gradient',
      'draw:frame',
      'draw:custom-shape'
    ];
  }
  */

  get keys() {
    return [
      'office:document-styles.office:font-face-decls.style:font-face',
      'office:document-styles.office:styles.draw:gradient',
      'office:document-styles.office:styles.draw:hatch',
      'office:document-styles.office:styles.draw:fill-image',
      'office:document-styles.office:styles.draw:marker',
      'office:document-styles.office:styles.draw:stroke-dash',
      'office:document-styles.office:styles.style:default-style',
      'office:document-styles.office:styles.style:style',
      /**
       * @warning `presentation-page-layout` causes OpenOffice to crash when opening the document
       */
      // 'office:document-styles.office:styles.style:presentation-page-layout',
      // 'office:document-styles.office:automatic-styles.style:page-layout',
      'office:document-styles.office:automatic-styles.style:style',
      'office:document-styles.office:automatic-styles.text:list-style',
      // 'office:document-styles.office:master-styles.draw:layer-set.draw:layer',
      // 'office:document-styles.office:master-styles.style:handout-master',
      'office:document-styles.office:master-styles.style:master-page'
    ];
  }

  // eslint-disable-next-line max-lines-per-function
  changeKeyReferences(content) {
    // let arr = extractArray(
    //   content,
    //   'office:document-styles.office:automatic-styles.style:style'
    // );
    // let mp = extractArray(
    //   content,
    //   'office:document-styles.office:master-styles.style:master-page'
    // );
    // mp.forEach(e => {
    //   delete e['presentation:notes'];
    //   delete e['draw:custom-shape']; // @temp
    //   extractArray(e, 'draw:custom-shape').forEach(s => {
    //     if (s['@_presentation:style-name']) {
    //       s['@_presentation:style-name'] =
    //         this.id + '_' + s['@_presentation:style-name'];
    //     }
    //   });
    // });
    // arr.forEach(e => {
    //   if (e['@_style:family'] === 'presentation') {
    //     e['@_style:name'] = this.id + '_' + e['@_style:name'];
    //   }
    // });

    return content;
  }
}
