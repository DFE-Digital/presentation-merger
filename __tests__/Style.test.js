import path from 'path';
import decompress from 'decompress';
import { toJson } from 'xml2json';
import Style from '../src/Style';

describe('Style', () => {
  let subject;
  beforeEach(async () => {
    const files = await decompress(
      path.join(__dirname, '__fixtures__/file1.odp')
    );
    const masterStylesFile = files.find(f => f.path === 'styles.xml');
    subject = new Style(JSON.parse(toJson(masterStylesFile.data)));
  });

  describe('namespaces', () => {
    it('returns the style XML namespaces for this presentation', () => {
      expect(subject.namespaces).toMatchSnapshot();
    });
  });
});
