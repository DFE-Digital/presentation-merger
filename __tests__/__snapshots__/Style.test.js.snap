// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Style masterPages returns master styles for the presenation 1`] = `undefined`;

exports[`Style namespaces returns the style XML namespaces for this presentation 1`] = `
Object {
  "xmlns:anim": "urn:oasis:names:tc:opendocument:xmlns:animation:1.0",
  "xmlns:chart": "urn:oasis:names:tc:opendocument:xmlns:chart:1.0",
  "xmlns:dc": "http://purl.org/dc/elements/1.1/",
  "xmlns:dom": "http://www.w3.org/2001/xml-events",
  "xmlns:dr3d": "urn:oasis:names:tc:opendocument:xmlns:dr3d:1.0",
  "xmlns:draw": "urn:oasis:names:tc:opendocument:xmlns:drawing:1.0",
  "xmlns:drawooo": "http://openoffice.org/2010/draw",
  "xmlns:fo": "urn:oasis:names:tc:opendocument:xmlns:xsl-fo-compatible:1.0",
  "xmlns:form": "urn:oasis:names:tc:opendocument:xmlns:form:1.0",
  "xmlns:grddl": "http://www.w3.org/2003/g/data-view#",
  "xmlns:math": "http://www.w3.org/1998/Math/MathML",
  "xmlns:meta": "urn:oasis:names:tc:opendocument:xmlns:meta:1.0",
  "xmlns:number": "urn:oasis:names:tc:opendocument:xmlns:datastyle:1.0",
  "xmlns:of": "urn:oasis:names:tc:opendocument:xmlns:of:1.2",
  "xmlns:office": "urn:oasis:names:tc:opendocument:xmlns:office:1.0",
  "xmlns:officeooo": "http://openoffice.org/2009/office",
  "xmlns:ooo": "http://openoffice.org/2004/office",
  "xmlns:oooc": "http://openoffice.org/2004/calc",
  "xmlns:ooow": "http://openoffice.org/2004/writer",
  "xmlns:presentation": "urn:oasis:names:tc:opendocument:xmlns:presentation:1.0",
  "xmlns:rpt": "http://openoffice.org/2005/report",
  "xmlns:script": "urn:oasis:names:tc:opendocument:xmlns:script:1.0",
  "xmlns:smil": "urn:oasis:names:tc:opendocument:xmlns:smil-compatible:1.0",
  "xmlns:style": "urn:oasis:names:tc:opendocument:xmlns:style:1.0",
  "xmlns:svg": "urn:oasis:names:tc:opendocument:xmlns:svg-compatible:1.0",
  "xmlns:table": "urn:oasis:names:tc:opendocument:xmlns:table:1.0",
  "xmlns:tableooo": "http://openoffice.org/2009/table",
  "xmlns:text": "urn:oasis:names:tc:opendocument:xmlns:text:1.0",
  "xmlns:xhtml": "http://www.w3.org/1999/xhtml",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
}
`;

exports[`Style presentationPageLayouts returns layouts for the presentation 1`] = `undefined`;

exports[`Style styles returns styles for the presentation slides 1`] = `Array []`;
